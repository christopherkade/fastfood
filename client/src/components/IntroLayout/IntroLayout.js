import React from "react"
import styled from "styled-components"

import { IntroCard } from "../IntroCard"
import data from "../../data/food.json"

const Wrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
`

const EmojiWrapper = styled.div`
  position: absolute;
  width: 100%;
  height: 100%;
  z-index: -1;
  display: grid;
  grid-gap: 10px;
  grid-template-columns: repeat(7, 1fr);
  align-items: center;
  justify-items: center;
  overflow: hidden;
`

const Emoji = styled.span`
  font-size: 3rem;
  display: inline-block;
  animation: scroll 20s infinite linear;

  @keyframes scroll {
    0% {
      transform: translateY(0px)
    }
    50% {
      transform: translateY(-500px)
    }
    100% {
      transform: translateY(0px)
    }
  }
`

const IntroLayout = ({ onClick }) => {
  const foodCopy = [...data.foods]

  // Add some more emojis to the mix
  foodCopy.forEach(food => {
    foodCopy.push(food)
  })

  return (
    <Wrapper>
      <EmojiWrapper>
        {foodCopy.map((food, index) => <Emoji key={index} role="img" aria-label="">{food.emoji}</Emoji>)}
      </EmojiWrapper>
      <IntroCard onClick={onClick} />
    </Wrapper>
  )
}

export default IntroLayout
