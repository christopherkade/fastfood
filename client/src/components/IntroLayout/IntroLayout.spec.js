import React from 'react'
import { render, cleanup } from 'react-testing-library'

import IntroLayout from "./IntroLayout"

afterEach(cleanup);

describe("<IntroLayout />", () => {
  describe("Snapshots", () => {
    it("should match snapshot", () => {
      const { container } = render(<IntroLayout onClick={() => { }} />);
      expect(container).toMatchSnapshot();
    });
  })
})
