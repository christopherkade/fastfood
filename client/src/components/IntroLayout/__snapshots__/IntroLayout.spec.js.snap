// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IntroLayout /> Snapshots should match snapshot 1`] = `
<div>
  <div
    class="sc-ifAKCX khZGyN"
  >
    <div
      class="sc-EHOje Gemmj"
    >
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🍚
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🍕
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🍔
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🍣
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🍟
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🌭
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🥓
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🥚
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🥞
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🍞
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🥨
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🧀
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🥗
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🥙
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🥪
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🌮
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🍖
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🍗
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🥩
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🦐
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🐟
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🦀
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🐙
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🦑
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🍅
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🍄
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🌽
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🥒
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🍆
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🥑
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🥔
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🥦
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🥕
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🥡
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🍱
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🍙
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🍛
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🍜
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🥟
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🍲
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🍝
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🥘
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🥣
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🥜
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🍿
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🥧
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🍦
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🍩
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🍪
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🎂
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🍰
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🍫
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🍬
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🍮
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🥝
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🍇
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🍉
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🍊
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🍌
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🍍
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🍎
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🍓
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🍑
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🍐
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🍒
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🥛
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        ☕
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🍵
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🥤
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🍺
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🍷
      </span>
      <span
        aria-label=""
        class="sc-bZQynM iNDdWj"
        role="img"
      >
        🍾
      </span>
    </div>
    <div
      class="sc-bdVaJa jOMnRk"
    >
      <h1
        class="sc-bwzfXH gyiady"
      >
        Fastfood
      </h1>
      <p
        class="sc-htpNat fnwEBe"
      >
        Argue less, eat more
      </p>
      <button
        class="sc-bxivhb vcWxY"
      >
        Let's go !
      </button>
    </div>
  </div>
</div>
`;
